model/riscv_pmp_regs.sail

```diff
@@ -154,15 +162,28 @@ function pmpReadCfgReg(n) = {
 }
 
 /* Helpers to handle locked entries */
-function pmpLocked(cfg: Pmpcfg_ent) -> bool =
+function pmpLockBit(cfg: Pmpcfg_ent) -> bool =
    cfg.L() == 0b1
+   
+function pmpLocked(cfg: Pmpcfg_ent) -> bool =
+   (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0)
 
 function pmpTORLocked(cfg: Pmpcfg_ent) -> bool =
-   (cfg.L() == 0b1) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
-
+   (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
+  
+/*Function implements 4b of Smepmp spec*/
+function checkExe(cfg: Pmpcfg_ent,v: bits(8)) -> Pmpcfg_ent =
+	 if ((mseccfg.MML() == 0b1) & (mseccfg.RLB() == 0b0) & (v[7] == bitone) & (v[2] == bitone) | (mseccfg.MML() == 0b1) & (mseccfg.RLB() == 0b0) & (v[7] == bitone) & (v[1] == bitone) | (mseccfg.MML() == 0b1) & (mseccfg.RLB() == 0b0) & (v[7] == bitone) & (v[1] == bitone) & (v[2] == bitone) | (mseccfg.MML() == 0b1) & (mseccfg.RLB() == 0b0) & (v[7] == bitone) & (v[0] == bitone) & (v[2] == bitone))
```

Comment:
This is not readable

Reply:
Condition is broken down for improved readability